java/ql/src/experimental/Security/CWE/CWE-522/InsecureBasicAuth.ql

```diff
@@ -66,25 +72,17 @@ class URLConstructor extends ClassInstanceExpr {
  * Gets a regular expression for matching private hosts.
  */
 private string getPrivateHostRegex() {
-  result = "localhost(/.*)?" or
-  result = "127\\.0\\.0\\.1(/.*)?" or // IPv4 patterns
-  result = "10(\\.[0-9]+){3}(/.*)?" or
-  result = "172\\.16(\\.[0-9]+){2}(/.*)?" or
-  result = "192.168(\\.[0-9]+){2}(/.*)?" or
-  result = "\\[0:0:0:0:0:0:0:1\\](/.*)?" or // IPv6 patterns
-  result = "\\[::1\\](/.*)?"
+  result = "(?i)localhost([:/].*)?|127\\.0\\.0\\.1([:/].*)?|10(\\.[0-9]+){3}([:/].*)?|172\\.16(\\.[0-9]+){2}([:/].*)?|192.168(\\.[0-9]+){2}([:/].*)?|\\[0:0:0:0:0:0:0:1\\]([:/].*)?|\\[::1\\]([:/].*)?" 
 }
 
 /**
  * String of HTTP URLs not in private domains.
  */
-class HttpString extends StringLiteral {
-  HttpString() {
+class HttpStringLiteral extends StringLiteral {
+  HttpStringLiteral() {
     // Match URLs with the HTTP protocol and without private IP addresses to reduce false positives.
     exists(string s | this.getRepresentedString() = s |
-      s.regexpMatch("(?i)http")
-      or
-      s.regexpMatch("(?i)http://.*") and
+      s.regexpMatch("(?i)http://[a-zA-Z0-9].*") and
```

Comment:
What was wrong with the previous regex (`(?i)http://.*`)? It appears the new regex would not match IPv6 addresses anymore.

Reply:
Thanks @Marcono1234 for reviewing the query for IPv6 addresses. I made the change not to match the prefix `http://:` only to eliminate false positives since hosts must not be in private domains to pass the test. I have thought a valid host must start with a letter or number.

I can just add `:` to the list of allowed characters. Does this suffice?

Reply:
> I made the change not to match the prefix `http://:` only to eliminate false positives since hosts must not be in private domains to pass the test.

Ah I did not know that something like `http://:8000` is possible (it appears it is equivalent to `localhost`). In that case including `[` would suffice I guess:
```suggestion
      s.regexpMatch("(?i)http://[\\[a-zA-Z0-9].*") and
```

Reply:
Agreed. Will do.

Reply:
I've added the `[` character as per the suggestion.